TheWarWithin/PriestShadow.lua

```diff
@@ -1530,6 +1534,14 @@ spec:RegisterAbilities( {
             if set_bonus.tier29_2pc > 0 then
                 addStack( "gathering_shadows" )
             end
+
+            if buff.voidheart.up and talent.darkening_horizon.enabled then
+                -- Extend Voidheart by 1 second, up to a maximum of 3 seconds during the active window
+                if void_blast_extension_count < 3 then  -- Ensure no more than 3 seconds extension
```

Comment:
First:  `voidheart` vs. `void_heart`.

This is going to clash in terms of real Void Bolt casts vs. predicted casts.  This needs to have a Combat Log event handler to count real Void Blast extensions and then `RegisterStateExpr` for a virtualized version that is reset to `nil` in `reset_precast`.

I think the code cruft in the Shaman modules for the Shadowlands Kyrian totem is probably the best example, but is more complicated than needed here.

For the CLEU handler, you'd need to watch for:
(Sub)Event | Reason
--- | ---
`SPELL_AURA_EXPIRES` - `voidheart` | Reset *actual* VB extension count to 0.
`SPELL_CAST_SUCCESS` - `void_bolt` | Increment VB extension count by 1; can cap at 3 but doesn't actually matter.
`???` | ???

For the state expression:
```Lua
spec:RegisterStateExpr( "vb_extensions", function()
    vb_extensions = vb_extension_count
    return vb_extensions
end )
```
and

```Lua
    -- Inside reset_precast:
    vb_extensions = nil
```

Then, in the handlers you can increment `vb_extensions` for the virtual game state and the value resets to the actual count every reset (and it's updated again with the *actual* casts of Void Bolt).

LMK if this makes sense.  Tracking what *real* data needs to be tracked vs. emulated data from predictions is basically the hardest part of working on this addon.  Since we need a count to persist beyond each set of recommendations, we need to collect the real information with CLEU (or other) events and then provide a way to interface/override the value during prediction sets.